tests/unit/test_multipan_rcp.cpp

```diff
@@ -0,0 +1,795 @@
+    VerifyOrQuit(instance != nullptr);
+
+    std::shared_ptr<TestNcp> ncp = std::make_shared<TestNcp>(instance);
+    TestHost                 host(ncp, SPINEL_HEADER_IID_0);
+
+    host.enableRawLink();
+
+    VerifyOrQuit(host.startTransmit() == SPINEL_STATUS_OK);
+    VerifyOrQuit(ncp->getPendingQueueSize() == 0);
+
+    /* Test that the response status is OK when transmit is already in progress
+       Test that requesting a transmit when already in progress enqueues the command */
+    VerifyOrQuit(host.startTransmit() == SPINEL_STATUS_OK);
+    VerifyOrQuit(ncp->getPendingQueueSize() == 1);
+
+    VerifyOrQuit(host.startTransmit() == SPINEL_STATUS_OK);
+    VerifyOrQuit(ncp->getPendingQueueSize() == 2);
+
+    /* Test that transmit command is dequeued when transmit is complete */
+    host.finishTransmit();
+    VerifyOrQuit(ncp->getPendingQueueSize() == 1);
+
+    host.finishTransmit();
+    VerifyOrQuit(ncp->getPendingQueueSize() == 0);
+
+    host.finishTransmit();
+    VerifyOrQuit(ncp->getPendingQueueSize() == 0);
+
+    testFreeInstance(instance);
+    printf("PASS\n");
+}
+
+void TestNcpBaseExceedPendingCommandQueueSize()
+{
+    printf("\tExceed Pending Command Queue Size - ");
+    ot::Instance *instance = testInitInstance();
+    VerifyOrQuit(instance != nullptr);
+
+    std::shared_ptr<TestNcp> ncp = std::make_shared<TestNcp>(instance);
+    TestHost                 host(ncp, SPINEL_HEADER_IID_0);
+
+    host.enableRawLink();
+
+    /* Test that the response status is OK and queue size increases until its maximum size */
+    for (size_t i = 0; i <= ncp->getMaxPendingQueueSize(); i++)
+    {
+        VerifyOrQuit(host.startTransmit() == SPINEL_STATUS_OK);
+        VerifyOrQuit(ncp->getPendingQueueSize() == i);
+    }
+
+    /* Test that the response status is NOMEM when requesting transmit with a full queue */
+    VerifyOrQuit(host.startTransmit() == SPINEL_STATUS_NOMEM);
+    VerifyOrQuit(ncp->getPendingQueueSize() == ncp->getMaxPendingQueueSize());
+
+    /* Test that queue size decreases from maximum size to empty */
+    for (size_t i = ncp->getMaxPendingQueueSize(); i > 0; i--)
+    {
+        VerifyOrQuit(ncp->getPendingQueueSize() == i);
+        host.finishTransmit();
+    }
+
+    VerifyOrQuit(ncp->getPendingQueueSize() == 0);
+    host.finishTransmit();
+    VerifyOrQuit(ncp->getPendingQueueSize() == 0);
+
+    testFreeInstance(instance);
+    printf("PASS\n");
+}
+
+void TestNcpBaseEnergyScanWithLinkRawDisabled()
+{
+    printf("\tEnergy Scan With Link Raw Disabled - ");
+    ot::Instance *instance = testInitInstance();
+    VerifyOrQuit(instance != nullptr);
+
+    std::shared_ptr<TestNcp> ncp = std::make_shared<TestNcp>(instance);
+    TestHost                 host(ncp, SPINEL_HEADER_IID_0);
+
+    host.disableRawLink();
+
+    /* Test that the response status is OK even though energy scan is skipped due to disabled link */
+    VerifyOrQuit((spinel_status_t)host.startEnergyScan() == SPINEL_STATUS_OK);
```

Comment:
Is this the correct return status even though the energy scan is skipped?

Reply:
That is the behavior of the property handler. As long as the link raw energy scan function does not fail, the NcpBase considers this to have an OK spinel status.

I am not sure how to feel about this, and I'm not sure it will matter much in practice, but it appears to be the expected behavior.